include/mbedtls/bignum.h

```diff
@@ -300,24 +304,28 @@ int mbedtls_mpi_safe_cond_assign( mbedtls_mpi *X, const mbedtls_mpi *Y, unsigned
  *
  * \param X        The first MPI. This must be initialized.
  * \param Y        The second MPI. This must be initialized.
- * \param assign   The condition deciding whether to perform
- *                 the swap or not. Possible values:
+ * \param swap     The condition deciding whether to perform
+ *                 the swap or not. Must be either 0 or 1:
  *                 * \c 1: Swap the values of \p X and \p Y.
  *                 * \c 0: Keep the original values of \p X and \p Y.
  *
  * \note           This function is equivalent to
- *                      if( assign ) mbedtls_mpi_swap( X, Y );
+ *                      if( swap ) mbedtls_mpi_swap( X, Y );
  *                 except that it avoids leaking any information about whether
- *                 the assignment was done or not (the above code may leak
+ *                 the swap was done or not (the above code may leak
  *                 information through branch prediction and/or memory access
  *                 patterns analysis).
  *
+ * \warning        If \p swap is neither 0 nor 1, the result of this function
+ *                 is indeterminate, and both \p X and \p Y might end up with
```

Comment:
How about
```suggestion
 *                 is undefined, and both \p X and \p Y might end up with
```
it is shorter and I think it conveys what we mean here as well.

Reply:
Well, I think that it is not entirely unlike C undefined. We absolutely give no guarantees whatsoever what will happen if the values are not one of these two. We don’t mean undefined behaviour in the compiler sense, but if the reader makes that mistake that still sends the message we want.

Also, this is a value being undefined, technically that is not the same as undefined behaviour, so it is not a complete overlap in terminology.

But then again, I don't mind leaving it as it is.

Reply:
In the C standard, the term for code that produces a valid result but there's no guarantee what this result is is _unspecified_ value. An _indeterminate_ value can be a trap representation, so the code may either produce an unspecified result or behave in some documented way in reaction to a trap. The adjective _undefined_ is reserved for behavior, where a program can do anything including launch demons out of your nose.

So here the C term would be _unspecified_. _Indeterminate_ isn't too far off. _Undefined_ would be misleading.

Reply:
I will use *unspecified* as the precisely correct term in the future; we can update these as and when we are "in the area"